views/wsprov_fetch_linked_objects.aql

```diff
@@ -4,12 +4,21 @@
 // Args:
 //   ws_ids - array of private workspace ids the user has access to
 //   obj_keys - wsprov_object key to find links for
+//   is_private - if present, limit to objects in workspaces that a user has access to
+//   is_public - if present, limit to objects in public workspaces
+//   *** if both is_private and is_public are true this will be treated as an OR ***
+//   types - simplified workspace types to return (e.g. Genome, Assembly)
 //   link_limit - number of link results (10-20 recommended)
 
-for o in wsprov_object
-  filter o._key in @obj_keys
-  for link in 1..10 any o wsprov_links options {uniqueVertices: 'global', bfs: true}
-    filter link.is_public || link.workspace_id IN @ws_ids
-    limit @link_limit
-    collect parent_id = o._id into links = link
-    return { links, parent_id }
+FOR o IN wsprov_object
+  FILTER o._key in @obj_keys
+  FOR v IN 1..10 ANY o wsprov_links
+    OPTIONS {uniqueVertices: 'global', bfs: true}
+    LET simple_type = SPLIT(SPLIT(v.ws_type, '-', 1)[0], '.')[1]
+    FILTER (@is_private && @is_public) ? (v.is_public || v.workspace_id IN @ws_ids) :
+        (!@is_private || v.workspace_id IN @ws_ids) && (!@is_public || v.is_public)
+    FILTER (!@types || simple_type IN @types)
+    LIMIT @link_limit
+    COLLECT parent_id = o._id into objects = v
+
+    RETURN { objects, parent_id }
```

Comment:
Can we make a couple changes to the return value here:
- Just `RETURN objects` so that I can use pagination. Don't need to collect on the parent_id anymore
- Remove the LIMIT @link_limit -- i can limit based on the batch_size update you made

Reply:
I'll also convert `@object_keys` to @`key` to reflect the fact that it will work on only one ref at a time